# ФОМП. МКР-1
Богомаз Р.Д.

- Метал: $Fe_{\alpha}$
- Тип ґратки: **ОЦК**
- Параметр ґратки, $a$: **`0.287 нм`**
- Радіус атому: **`0.156 нм`**
- Модуль зсуву $G$: **`82.5 ГПа`**
- Коеф. Пуассона $\mu$:** `0.29`**
- Довжина дислокації $L=N \times a$: **`300 нм`**
- Дислокація: $\frac{a}{2}[111]$
- Щільність дислокації $\rho$: $3 \times 10^9 \text{см}^2$ 
- Конц. розч. атомів $C$: **`0.3` ат.%**
- Розмір зерна:** `170 мкм`**
- Розмір частинки: **`20 нм`**
- Енергія дефекта пакування: **140 **$\frac{\text{ерг}}{\text{см}^2}$
- Відстань від поверхні/між частинками $L=N \times a$: **`100 нм`**
- Товщина покриття: **`120 нм`**

In [1]:
from math import sqrt, log2, exp, pi

a = 0.287e-9
r = 0.156e-9
G = 82.5e9
mu = 0.29
L_dislocation = 300e-9
rho = 3e9/1e-4
C = 0.3e-2
grain = 170e-6
part = 20e-9
Edefect = 140e-3
L = 100e-9
d = 120e-9

h = 1
k = 1
l = 1

### Оцінити діаметр ядра крайової дислокації в матеріалі, якщо період гратки $a$.
(1 бал)
$$d_{\text{ядра}} \approx 3 \left| \bar{b} \right| = 3 \frac{a}{2}\sqrt{h^2+k^2+l^2}$$

In [2]:
b = a / 2 * sqrt(h**2 + k**2 + l**2)
d_core = 3 * b
"{:.2e} м".format(d_core)

'7.46e-10 м'

### Розрахувати енергію повної одиничної гвинтової дислокації в матеріалі, якщо всі дані наведено в таблиці
(2 бали)
$$E_{\textit{гв}} = \alpha_{\textit{гв}}~G~\left|\bar{b}\right|^2~L ,$$
де $\alpha_{\textit{гв}} = 0.37$

In [3]:
E_screw = 0.37 * G * b**2 * L_dislocation
"{:.2e} Дж".format(E_screw)

'5.66e-16 Дж'

### Розрахувати напруження руху дислокації (силу Пайєрлса-Набарро) для заданого матеріалу.
(3 бали)
$$\tau_{\textit{П-Н}} = \frac{2G}{1-\mu}\exp\frac{-2\pi a}{b(1-\mu)}$$

In [13]:
tau_pn = 2 * G / (1-mu) * exp((-2*pi*a)/(b*(1-mu)))
"{:.2e} Па".format(tau_pn)

'8.48e+06 Па'

### Оцінити кількість домішкових атомів, необхідну для утворення насичених атмосфер Котрелла в деформованому матеріалі, якщо період гратки $a$, густина дислокацій $\rho$.
(3 бали)
$$C_{\textit{н}} = \rho a^2$$

In [5]:
C_sat = rho * a**2
print("Концентрація насичення:  {:.2e}\nКонц-я насичених атомів: {:.2e}".format(C_sat, C))

Концентрація насичення:  2.47e-06
Конц-я насичених атомів: 3.00e-03


### Оцініть силу взаємодії дислокації з поверхнею та опишіть рух дислокації під дією цієї сили.
Матриця – ваш варіант на контрольній роботі. Покриття – матеріал наступного варіанту.

(4 бали)
```
```
Покриття: $Cu, G = 45~\text{ГПа}$

Оскільки $G_{\textit{матеріал}} > G_{\textit{покриття}}$, то дислокація притягується до поверхні розподілу і може перейти у інше середовище. Розраховуємо силу взаємодії дислокації з поверхнею за формулою:
$$F = \frac{G b^2}{4\pi L}$$

In [6]:
F = G * b**2 / (4* pi * L)
"{:.2e} Н".format(F)

'4.06e-03 Н'

### Розрахувати зміцнення внаслідок наявності в матеріалі когерентних часток якщо їх об’ємна частка $f$
(матеріал 2 (варіант $N-1$) ), 

($f=C$ попередньої задачі).

(4 бали)
```
```

- Матеріал: $Mn_{\alpha}$
- $f = 2.47\times10^6$
- $a_b = 0.89~\text{нм}$

При наявності в матеріалі зон Гіньє-Престона та/або когерентних часток
зміцнення розраховується за виразом, запропонованим Моттом і Набарро:

$$\Delta\sigma = 2 G_m \varkappa^{\frac{3}{2}} (r_{\text{ч}}~f~b_m^{-1})^{\frac{1}{2}},$$

$$\varkappa = \frac{3K\varepsilon}{3K+2E(1+\mu)},$$

$$\varepsilon = 2 \frac{a_m-a_b}{a_b + a_m},$$

де $G_m$ – модуль зсуву матриці, $f$ – об’ємна доля частинок в матеріалі, $\varepsilon$ – параметр
невідповідності, $a_m$ та $a_b$ – параметри граток матриці і виділення, $K$ - об'ємний
модуль пружності виділення, $\mu$ - коефіцієнт Пуассона матриці.

$$K = \frac{2G(1+\mu)}{3(1-2\mu)}$$

$$E = 2G(1+\mu)$$

In [7]:
f = C_sat
a_b = 0.89e-9
K = 2 * G * (1 + mu) / (3*(1-2*mu))
E = 2 * G * (1 + mu)
epsilon = 2 * (a_b - a) / (a_b + a)
kappa = 3 * K * epsilon / (3 * K + 2 * E * (1+ mu))
Dsigma = 2 * G * kappa**(3/2) * ((r * f * b**(-1))**(0.5))
"{:.2e} Па".format(Dsigma)

'7.09e+07 Па'

### Розрахуйте зернограничне зміцнення матеріалу, якщо розмір зерна $d$. 
Різниця факторів Шміда: для ОЦК - 0,34, для ГЦК – 0,2.

(4 бали)
```
```
Розрахувати додаткове деформаційне зміцнення полікристалу можна за
допомогою формули:

$$\Delta\sigma_{\textit{зг}} = 3 M~\Delta m~\tau_0~\sqrt{\frac{r_0}{d}},$$
де $М$ – орієнтаційний фактор Тейлора (для ГЦК гратки – 3,1; для ОЦК гратки –
2,0); $\Delta m$ – усереднений фактор орієнтації (може змінюватися від 0 до 0,5); $\tau_0$ –
критичне дотичне напруження для монокристалів; $r_0$ – деяка константа, що має
розмірність довжини, становить 2-4 мкм; $d$ – розмір зерна.

In [8]:
M = 3.1
Deltam = 0.34

Delta_sigma_grain = 3 * M * Deltam * tau_pn * sqrt(part/d)
"{:.2e} Па".format(Delta_sigma_grain)

'1.74e+09 Па'

### Розрахуйте ширину розтягнутої дислокації в матеріалі. 
(4 бали)
```
```
В ОЦК гратці розтягнута дислокація формується за формулою:

$$\frac{a}{2}[\bar{1}\bar{1}1] = \frac{a}{6}[\bar{1}\bar{1}1] + \frac{a}{3}[\bar{1}\bar{1}1]$$

В ГЦК гратці розтягнута дислокація формується за формулою:

$$\frac{a}{2}[110] = \frac{a}{6}[112] + \frac{a}{3}[112]$$

In [9]:
def vector_b(n, a, h, k, l):
    return a/n * sqrt(h**2 + k**2 + l**2)

b_1 = vector_b(2, a, -1, -1, 1)**2
b_2 = vector_b(6, a, -1, -1, 1)**2
b_3 = vector_b(3, a, -1, -1, 1)**2

print("Енергія до:    {:.2e} \nЕнергія після: {:.2e}".format(b_1, b_2+b_3))

Енергія до:    6.18e-20 
Енергія після: 3.43e-20


Реакція утворення дислокації пройде, тому розраховуємо ширину розтягнутої дислокації:

$$d_0=K \frac{G~b_2~b_3}{\gamma},$$

де $K$ – стала, $G$ – модуль зсуву, $b_2$ та $b_3$ – вектори Бюргерса дислокацій Шоклі, $\gamma$ –
поверхневий натяг дефекту пакування.

In [11]:
b_2 = vector_b(6, a, -1, -1, 1)
b_3 = vector_b(3, a, -1, -1, 1)

const = 0.9
d_0 = const * (G * b_2 * b_3)/Edefect
"{:.2e} м".format(d_0)

'7.28e-09 м'